### DISCUSSION TITLE
problem deploy - action ref branch name

### DISCUSSION THREAD

**Discussion by LeonanRodrigues** [community/community#67624] [Pull Requests]
### Select Topic Area

Bug

### Body

Dear,

We are unable to carry out the deploy.
When we try we receive the following error:
**_"The branch "refs/heads/develop" is not allowed to deploy to develop due to environmental protection rules"_**

We noticed that the branch previously only appeared with the main name (develop, staging or master) and is now appearing with the full path (/refs/heads/$branch).

It is worth noting that nothing has changed on our side.
Can you help us?

**Comment by dcarvalhal** [comment]
I believe it may be related to the resolution of the previous incident.

**Comment by dcarvalhal** [comment]
https://www.githubstatus.com/incidents/00lscqwb6ht5

**Comment by dcarvalhal** [comment]
![image](https://github.com/community/community/assets/36567989/7fdf8b21-f027-43a6-b366-39a251911b8f)

**Comment by LeonanRodrigues** [comment]
update

**Comment by LeonanRodrigues** [comment]
```
run: echo "branch=$(echo ${GITHUB_REF#refs/heads/})" >> $GITHUB_OUTPUT
id: extract_branch
```

**Comment by timcole** [comment]
We're currently experiencing the same issue. It worked a few hours ago but recently the branch is refs/heads/main instead of just main.
![](https://t.pics/4Yo0AXs7LnM.png)

**Comment by gzt-fastradius** [comment]
We are seeing a similar issues.  Our Github Action workflows aren't working because all pushes to `main` show up as `refs/heads/main` and therefore our workflow ignores it due to filtering we have in place.

**Comment by allwalte** [comment]
Same issue here, same timing (last couple of hours), and same presentation, with the branch seeming to read as refs/heads/main instead of just main and therefore failing protection rules.  Also no changes on our end in that timeframe.

**Comment by ch4rms** [comment]
Broke all of our deploys

**Reply by ch4rms** [reply]
For now I have to disable all deployment protection, by disabling the protection we have on the repo managing GH via TF :trollface:

**Reply by moloch--** [reply]
Same here

**Comment by ch4rms** [comment]
Related? https://github.blog/changelog/2023-09-13-github-actions-updates-to-github_ref-and-github-ref/

**Reply by allwalte** [reply]
That was very much my hunch as well; some sort of intended side effect or consequence?

**Reply by ch4rms** [reply]
If I select `main` on a workflow_dispatch style workflow, main is protected, and the environment protection is set to `protected branches`, it is blocked :(

**Comment by LeonanRodrigues** [comment]
Validate again guys, it looks like it's back to normal

@timcole @moloch-- @ch4rms @gzt-fastradius @allwalte

**Reply by Jeff-Dyck-SOF** [reply]
Yup, looks resolved, thanks. I've successfully deployed with Selected Branches set.

**Reply by ch4rms** [reply]
:man_facepalming: Just removed env deploy protection on 50 repos....

**Reply by LeonanRodrigues** [reply]
@ch4rms back protection, man

**Reply by ch4rms** [reply]
Thanks @LeonanRodrigues for the follow up and tags!

**Reply by gzt-fastradius** [reply]
Yup, can confirm
